sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -701,11 +703,15 @@ def _plot_one_way_partial_dependence(
             )
 
         if self.kind in ("average", "both"):
-            label = None if self.kind == "average" else "average"
+            # do not modify the `line_kw` in place since it might be reuse for
+            # each feature for which the partial dependence is requested
+            line_kw = deepcopy(line_kw)
```

Comment:
Do you think this is cleaner:

```py
default_line_kw = {'label': None if self.kind == "average" else "average"}
line_kw = {**default_line_kw, **line_kw}
```

And then update the signature of `_plot_average_dependence` to only accept `line_kw`.

This would lower the number of arguments for `_plot_average_dependence` as well.

Reply:
Sounds like a good suggestion.

Reply:
Nice